File: src/csharp/Foundgine.Core/Interfaces/AuthorizationOperationName.cs
namespace Foundgine.Core.Abstractions;

/// <summary>
/// Optional named refinement of an <see cref="AuthorizationOperation"/> for
/// policies that distinguish domain-specific write intents (for example
/// "Invoice.Pay" versus "Invoice.Update") beyond the coarse Read/Write gate.
/// This is a policy-facing hint only: it never changes what
/// <see cref="AuthorizationOperation"/> means structurally, and it carries no
/// storage or provider semantics.
/// </summary>
public readonly record struct AuthorizationOperationName(string Value)
{
    public override string ToString() => Value;
}

File: src/java/foundgine-core/src/main/java/com/foundgine/core/abstractions/AuthorizationOperationName.java
package com.foundgine.core.abstractions;

/**
 * Port of {@code Foundgine.Core.Abstractions.AuthorizationOperationName}.
 *
 * <p>
 * Optional named refinement of an {@link AuthorizationOperation} for policies
 * that distinguish domain-specific write intents (for example "Invoice.Pay"
 * versus "Invoice.Update") beyond the coarse Read/Write gate. This is a
 * policy-facing hint only: it never changes what {@link AuthorizationOperation}
 * means structurally, and it carries no storage or provider semantics.
 */
public record AuthorizationOperationName(String value) {

	@Override
	public String toString() {
		return value;
	}
}
